talisker/endpoints.py

```diff
@@ -247,7 +260,221 @@ def metrics(self, request):
+
+        sorted_master = [(k, master[k]) for k in MASTER_FIELDS if k in master]
+
+        return html_response(
+            'Workers',
+            ['<h2>Workers</h2>'],
+            table(rows, headers=HEADERS),
+            ['<h2>Process Information</h2>'],
+            table(sorted_master),
+            ['<h2>Process Environment (whitelist)</h2>'],
+            table(clean_environ),
+        )
+
+    @private
+    def objgraph(self, request):
+        import objgraph
+        limit = int(request.args.get('limit', 10))
+        types = objgraph.most_common_types(limit=limit, shortnames=False)
+        leaking = objgraph.most_common_types(
+            limit=limit,
+            objects=objgraph.get_leaking_objects(),
+            shortnames=False,
+        )
+
+        limits = '<p>Number of items: {}, {}, {}</p>'.format(
+            link('{}', request.path + '?limit={}', 10),
+            link('{}', request.path + '?limit={}', 20),
+            link('{}', request.path + '?limit={}', 50),
+        )
+        return html_response(
+            'Python Objects',
+            ['<h1>Python Objects for Worker pid {}</h1>'.format(os.getpid())],
+            [limits],
+            ['<h2>Most Common Objects</h2>'],
+            table(types),
+            ['<h2>Leaking Objects (no referrer)</h2>'],
+            table(leaking),
+        )
+
+
+# diy html templating
+CDN = '//cdnjs.cloudflare.com/ajax/libs'
+HEADER = """
+<head>
+    <title>Talisker: {title}</title>
+    <link rel="stylesheet"
+    href="{cdn}/twitter-bootstrap/4.1.1/css/bootstrap.min.css">
+    <link rel="stylesheet"
+    href="{cdn}/bootstrap-table/1.12.1/bootstrap-table.min.css">
+    <script
+    src="{cdn}/jquery/3.3.1/jquery.js"></script>
+    <script
+    src="{cdn}/twitter-bootstrap/4.1.1/js/bootstrap.min.js"></script>
+    <script
+    src="{cdn}/bootstrap-table/1.12.1/bootstrap-table.min.js"></script>
+</head>
+""".format(cdn=CDN, title='{title}')
+
+
+def html_response(title, *iters):
+    head = HEADER.format(title=title)
+    body = chain(head, *iters)
+    return Response(body, mimetype='text/html')
+
+
+def table(data, headers=None):
+    css_class = "table table-striped table-hover table-bordered"
+    yield '<table class="{}">'.format(css_class)
+    if headers is not None:
+        yield '<thead><tr>'
+        for header in headers:
+            yield '<th>{}</th>'.format(header)
+        yield '</tr></thead>'
+    yield '<tbody>'
+    for row in data:
+        yield '<tr>'
+        for col in row:
+            yield '<td>{}</td>'.format(col)
+        yield '</tr>'
+    yield '</tbody>'
+    yield '</table>'
+
+
+def ul(items):
+    yield '<ul>'
+    for item in items:
+        yield '<li>{}</li>'.format(item)
+    yield '</ul>'
+
+
+def link(text, href, *args, **kwargs):
+    return '<a href="{}">{}</a>'.format(
+        href.format(*args, **kwargs),
+        html.escape(text.format(*args, **kwargs)),
+    )
+
+
+MASTER_FIELDS = [
+    'cwd',
+    'cmdline',
```

Comment:
I guess there *shouldn't* be anything sensitive in the process command line, but it's not unheard of to perpetrate quick hacks there that then stick around.  Have you audited our existing services for sensitive information on command lines?

Reply:
So, for gunicorn, we do things via the charm layers, which don't use the cli to configure, we use config files. 

In theory, you could use --env  to pass secrets, or perhaps it could leak info about SSL config if you use that via the cli

Reply:
Hmm, I should check SCA